packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1243,13 +1313,13 @@ export function hydrateInstance(
   const isConcurrentMode =
     ((internalInstanceHandle: Fiber).mode & ConcurrentMode) !== NoMode;
 
-  let parentNamespace;
+  let parentNamespace: string | null;
   if (__DEV__) {
     const hostContextDev = ((hostContext: any): HostContextDev);
-    parentNamespace = hostContextDev.namespace;
+    parentNamespace = getNamespaceURIFromHostContext(hostContextDev.context);
   } else {
     const hostContextProd = ((hostContext: any): HostContextProd);
-    parentNamespace = hostContextProd;
+    parentNamespace = getNamespaceURIFromHostContext(hostContextProd);
```

Comment:
This is useless in prod because it's just going to pass it to a DEV only property.

It used to be a noop since it's just an alias but now that's a little different.

Might as well just pass the enum down though because that call below is a just a simple comparison so it can just compare against the enum.